pddl/parser/problem.py

```diff
@@ -143,37 +139,10 @@ def goal(self, args):
         """Process the 'goal' rule."""
         return "goal", args[2]
 
-    def gd_binary_comparison(self, args):
-        """Process the 'gd' comparison rule."""
-        left = args[2]
-        right = args[3]
-        if args[1] == Symbols.GREATER_EQUAL.value:
-            return GreaterEqualThan(left, right)
-        elif args[1] == Symbols.GREATER.value:
-            return GreaterThan(left, right)
-        elif args[1] == Symbols.LESSER_EQUAL.value:
-            return LesserEqualThan(left, right)
-        elif args[1] == Symbols.LESSER.value:
-            return LesserThan(left, right)
-        elif args[1] == Symbols.EQUAL.value:
-            return FunctionEqualTo(left, right)
-        else:
-            raise PDDLParsingError(f"Unknown comparison operator: {args[1]}")
-
-    def gd_name(self, args):
+    def gd(self, args):
         """Process the 'gd_name' rule."""
-        if len(args) == 1:
-            return args[0]
-        elif args[1] == Symbols.NOT.value:
-            return Not(args[2])
-        elif args[1] == Symbols.AND.value:
-            return And(*args[2:-1])
-        elif args[1] == Symbols.OR.value:
-            return Or(*args[2:-1])
-        elif args[1] in BINARY_COMP_SYMBOLS:
-            return self.gd_binary_comparison(args)
-        else:
-            raise ParseError
+        result = self._domain_transformer.gd(args)
+        return result
```

Comment:
The parsing of the `gd` term now relies on the Transformer class for the PDDL domain.